### ISSUE TITLE
NullPointerException caused by collapsed attribute from p:panel (PF 10, WebSphere)

### ISSUE THREAD

**Issue by pneumcke** [primefaces/primefaces#8152]
We have a p:panel in our main template that has a collapsed attribute with an EL expression.

This works fine when the page is rendered for the first time. A NullPointerException appears in the update phase:

`
javax.servlet.ServletException: java.lang.NullPointerException
	at org.omnifaces.filter.FacesExceptionFilter.doFilter(FacesExceptionFilter.java:105)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:197)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:90)
	at org.omnifaces.filter.GzipResponseFilter.doFilter(GzipResponseFilter.java:181)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:197)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:90)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:969)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1109)
	at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:82)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:963)
	at com.ibm.ws.webcontainer.WSWebContainer.handleRequest(WSWebContainer.java:1817)
	at com.ibm.ws.webcontainer.channel.WCChannelLink.ready(WCChannelLink.java:382)
	at com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:465)
	at com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.handleNewRequest(HttpInboundLink.java:532)
	at com.ibm.ws.http.channel.inbound.impl.HttpInboundLink.processRequest(HttpInboundLink.java:318)
	at com.ibm.ws.http.channel.inbound.impl.HttpICLReadCallback.complete(HttpICLReadCallback.java:88)
	at com.ibm.ws.tcp.channel.impl.AioReadCompletionListener.futureCompleted(AioReadCompletionListener.java:175)
	at com.ibm.io.async.AbstractAsyncFuture.invokeCallback(AbstractAsyncFuture.java:217)
	at com.ibm.io.async.AsyncChannelFuture.fireCompletionActions(AsyncChannelFuture.java:161)
	at com.ibm.io.async.AsyncFuture.completed(AsyncFuture.java:138)
	at com.ibm.io.async.ResultHandler.complete(ResultHandler.java:204)
	at com.ibm.io.async.ResultHandler.runEventProcessingLoop(ResultHandler.java:775)
	at com.ibm.io.async.ResultHandler$2.run(ResultHandler.java:905)
	at com.ibm.ws.util.ThreadPool$Worker.run(ThreadPool.java:1909)
Caused by: java.lang.NullPointerException
	at org.apache.el.parser.AstNot.getValue(AstNot.java:45)
	at org.apache.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:184)
	at org.jboss.weld.el.WeldValueExpression.getValue(WeldValueExpression.java:50)
	at org.apache.myfaces.view.facelets.el.ContextAwareTagValueExpression.getValue(ContextAwareTagValueExpression.java:96)
	at org.primefaces.el.ValueExpressionAnalyzer.intercept(ValueExpressionAnalyzer.java:109)
	at org.primefaces.el.ValueExpressionAnalyzer.toValueReference(ValueExpressionAnalyzer.java:119)
	at org.primefaces.el.ValueExpressionAnalyzer.getExpression(ValueExpressionAnalyzer.java:69)
	at org.primefaces.component.panel.Panel.processUpdates(Panel.java:144)
	at javax.faces.component.UIComponentBase.processUpdates(UIComponentBase.java:1505)
	at javax.faces.component.UIComponentBase.processUpdates(UIComponentBase.java:1505)
	at javax.faces.component.UIViewRoot._processUpdatesDefault(UIViewRoot.java:1694)
	at javax.faces.component.UIViewRoot.access$700(UIViewRoot.java:77)
	at javax.faces.component.UIViewRoot$UpdateModelPhaseProcessor.process(UIViewRoot.java:1832)
	at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1655)
	at javax.faces.component.UIViewRoot.processUpdates(UIViewRoot.java:916)
	at org.apache.myfaces.lifecycle.UpdateModelValuesExecutor.execute(UpdateModelValuesExecutor.java:38)
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:196)
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:143)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:198)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1235)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:779)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:478)
	at com.ibm.ws.webcontainer.servlet.ServletWrapperImpl.handleRequest(ServletWrapperImpl.java:179)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:143)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:96)
	at com.wincor.adminconsole.jsf.filter.PrimeFacesDownloadCookieFilter.doFilter(PrimeFacesDownloadCookieFilter.java:59)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:197)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:90)
	...
`

Maybe related to https://github.com/primefaces/primefaces/issues/7738?

Would it be possible to make the changes available in 10.0.9?

**Environment:**
 - PF Version: _10.0.8
 - Application server: _Websphere classic 9.0.5.1_
 - JSF + version: _Apache MyFaces / JSF 2.2_
 - Affected browsers: _ALL_

The same code works fine with WLP and JBoss.

**Example XHTML**
```html
<p:panel  collapsed="#{not XXXBean.wsmMonitorVisible}" toggleable="true" ...
```

**Example Bean**
```java
    public boolean isWsmMonitorVisible() {
        return wsmMonitorVisible;
    }

    public void setWsmMonitorVisible(boolean wsmMonitorVisible) {
        return this.wsmMonitorVisible = wsmMonitorVisible;
    }
}
```

**Comment by tandraschko** [comment]
yep, thats a duplicate
i can only ping @mertsincan to put this into ELITE
otherwise try PrimeFaces PRO or use 11.0.0-RC2

**Comment by pneumcke** [comment]
We just migrated to PF 10 - switching to PF 11 would require a new round of changes to the complete code page. 
I would really love to have this fix in the next maintenance release of 10.0.x
Thanks a lot!

**Comment by tandraschko** [comment]
i really suggest to use 11.0 - it is much more stable as 10.0 as we introduced many integration tests
also it doesnt contain many breaking changes
